### ISSUE TITLE
setting state inside `onSelectStart` disables date range selection

### ISSUE THREAD

**Issue by nachikethashu** [onefinestay/react-daterange-picker#203]
Here's the link to the issue reproduced in codesandbox - https://codesandbox.io/s/vjq31pov87

Click on any date and it'll never go into range selection mode. keeps getting stuck in from-date selection.

All I'm doing there is setting some state inside(not related to range) `onSelectStart` handler.

Is it a bug from the component or my logic?

**Comment by nachikethashu** [comment]
Hello,
Can you confirm whether it's a bug or not?

**Comment by AlanFoster** [comment]
Hi @nachikethashu - Could you clarify what you would like to happen?

Do you want multiple selected ranges?

**Comment by nachikethashu** [comment]
No, If you check this -  https://codesandbox.io/s/vjq31pov87 you can see that the range selection is not working here. 

It works only if you remove `setState` inside `handleDateRangeSelectStart`

**Comment by abaune** [comment]
@nachikethashu ran into this issue as well. Was able to have a quick work around with 
`onSelectStart={() => this.handleDateRangeSelectStart}`
